### PR TITLE
add Timer

### PR DESCRIPTION
**Pull request by ostanmadalina92**

### CODE REVIEW

File: src/components/Timer/Timer.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import { TimerWrapper, TimerH3, TimerH1, TimerRectangle } from "./Timer.style";
+import rectangle1 from "./assets/Rectangle.svg";
+import rectangle2 from "./assets/Rectangle2.svg";
+
+import { useState, useEffect } from "react";
+
+export interface ITimer {
+  player: string;
+  time: number;
```
Comment (razwan) [inline src/components/Timer/Timer.tsx:10]: `time` doesn't seem to be used anywhere. Might as well just remove it

File: src/components/Timer/Timer.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  const [timeLeft, setTimeLeft] = useState(30);
+
+  useEffect(() => {
+
+    if (timeLeft === 0){ return }
+    const interval = setInterval(() => {
+      setTimeLeft(() => timeLeft - 1);
+    }, 1000);
+    return () => clearInterval(interval);
+  },[timeLeft]);
```
Comment (razwan) [inline src/components/Timer/Timer.tsx:25]: This section of code could use some white space for more readability

File: src/components/Timer/Timer.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+const Timer: React.FC<ITimer> = (props) => {
+  const { player } = props;
+
+  const [timeLeft, setTimeLeft] = useState(30);
+
+  useEffect(() => {
+
+    if (timeLeft === 0){ return }
+    const interval = setInterval(() => {
+      setTimeLeft(() => timeLeft - 1);
```
Comment (razwan) [inline src/components/Timer/Timer.tsx:22]: This should be
```js
setTimeLeft(() => timeLeft - 1);
```

Comment (razwan) [inline src/components/Timer/Timer.tsx:22]: This should be
```js
setTimeLeft(timeLeft => timeLeft - 1);
```
This way the setInterval would update the state with the proper value, and you won't need `timeLeft` in the dependency array anymore

**Review by razwan — CHANGES_REQUESTED** [review]
Hey @ostanmadalina92! Nice job so far!

I left some comment for you to address, and also, please render the Timer component in the app footer and make styles adjustments if necessary.

**Review by veresbianca — APPROVED** [review]

File: src/components/Timer/Timer.style.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import styled from "styled-components";
+import { theme } from "../../styles/Theme.style";
+import { LargeHeading, SmallHeading } from "../Heading/Heading.style";
+
+export const TimerWrapper = styled.div`
+  text-align: center;
+  position: relative;
+`;
+
+export const TimerRectangle = styled.img`
```
Comment (FedoraBianca) [inline src/components/Timer/Timer.style.tsx:10]: There's no need to create 'TimerRectangle' if it doesn't add any styles. You can simply use an img element.

File: src/components/Timer/Timer.tsx [hunk 5]
```diff
@@ -17,16 +16,20 @@ const Timer: React.FC<ITimer> = (props) => {
...
+
     const interval = setInterval(() => {
-      setTimeLeft(() => timeLeft - 1);
+
+      setTimeLeft(timeLeft => timeLeft - 1);
     }, 1000);
+
     return () => clearInterval(interval);
+
   },[timeLeft]);
```
Comment (razwan) [inline src/components/Timer/Timer.tsx:28]: Not needed anymore

**Review by razwan — APPROVED** [review]

**Pull request merged**